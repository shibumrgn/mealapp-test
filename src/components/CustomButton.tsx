import {Text, Pressable, StyleSheet} from 'react-native';
import React from 'react';
import {Brand} from '../constants/Brand';

interface IProps {
    title: string;
    isHighlighted: boolean;
    onClick?: () => void;
}

const CustomButton = (props: IProps) => {
    const styles = StyleSheet.create({
        parent: {
            marginEnd: 5,
        },
        title: {
            backgroundColor: props.isHighlighted
                ? Brand.accentColor
                : Brand?.primaryColor,
            padding: 10,
            borderRadius: 10,
            color: !props.isHighlighted
                ? Brand.accentColor
                : Brand?.primaryColor,
            fontWeight: 'bold',
        },
    });
    return (
        <Pressable style={styles.parent} onPress={props.onClick}>
            <Text style={styles.title}>{props.title}</Text>
        </Pressable>
    );
};

export default CustomButton;
