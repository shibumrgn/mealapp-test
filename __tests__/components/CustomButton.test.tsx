
import 'react-native';
import React from 'react'; 
import renderer from 'react-test-renderer';
import CustomButton from '../../src/components/CustomButton';
 
 it('renders correctly', () => {
   const render = renderer.create(<CustomButton title='Test' isHighlighted/>)?.toJSON();
   expect(render).toMatchSnapshot();
 });
 