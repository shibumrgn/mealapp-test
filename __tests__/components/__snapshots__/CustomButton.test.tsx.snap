// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginEnd": 5,
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "#11e0c4",
        "borderRadius": 10,
        "color": "#ffffff",
        "fontWeight": "bold",
        "padding": 10,
      }
    }
  >
    Test
  </Text>
</View>
`;
